### PR TITLE
BFB-414: WaveSandbox Level Greybox and Basic Navmesh

### PR DESCRIPTION
**Pull request by csweet22**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-414

**Description**
Created the greybox and intial navmesh for the wave-based sandbox level.

**Screenshots / GIFs**
image

### CODE REVIEW

File: Assets/EnemyBulletProperties.asset [hunk 1]
```diff
@@ -0,0 +1,18 @@
+%YAML 1.1
```
Comment (m4rtint) [inline Assets/EnemyBulletProperties.asset:1]: Can you place this in the correct folder

File: Assets/_BForBoss/_Entities/Enemies/FlyingRobot/Scripts/OffMeshLinkChecker.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        private void AcquireOffMeshLink()
+        {
+            if (_link != null)
+            {
+                return;
+            }
+            
+            _link = _agent.currentOffMeshLinkData.offMeshLink;
+            _oldCost = _link.costOverride;
+            _link.costOverride = _oldCost <= 0.0f ? (_multiplier * 1) : _oldCost * _multiplier;
```
Comment (m4rtint) [inline Assets/_BForBoss/_Entities/Enemies/FlyingRobot/Scripts/OffMeshLinkChecker.cs:36]: `_oldCost <= 0.0f ? (_multiplier * 1) : _oldCost * _multiplier;` 

Double check this for any redundancies.

File: Assets/EnemyBulletProperties.asset (hunk 1 above)
Comment (csweet22) [reply Assets/EnemyBulletProperties.asset:1]: Can do! I moved it to _Weapons > Scripts > ScriptableObjects > Bullets

File: Assets/_BForBoss/_Entities/Enemies/FlyingRobot/Scripts/OffMeshLinkChecker.cs (hunk 2 above)
Comment (csweet22) [reply Assets/_BForBoss/_Entities/Enemies/FlyingRobot/Scripts/OffMeshLinkChecker.cs:36]: I just copied this code straight from Gursaanj, but I can change the (_multiplier * 1) to just _multiplier if that is what you are referring to

Comment (m4rtint) [reply Assets/_BForBoss/_Entities/Enemies/FlyingRobot/Scripts/OffMeshLinkChecker.cs:36]: yup. If you see anything off, don't be afraid to update it.

Comment (csweet22) [reply Assets/_BForBoss/_Entities/Enemies/FlyingRobot/Scripts/OffMeshLinkChecker.cs]: Sounds good! It has been changed 👍

**Comment by m4rtint** [comment]
image
Can you move the Health pick up prefab to the correct place in the hierachy

**Review by m4rtint — APPROVED** [review]
lgtm, only one single change left

**Comment by m4rtint** [comment]
image
you don't have to do this now, but in the future, if you're changing something, just change it with the proper naming/grouping. 
Even though ColeAdditions are to separete the Original from yours, down the line it can get confusing as to why it's separated.

**Comment by m4rtint** [comment]
And can you change the name of "Entities" to "Interactables"

**Comment by csweet22** [comment]
> you don't have to do this now, but in the future, if you're changing something, just change it with the proper naming/grouping. Even though ColeAdditions are to separete the Original from yours, down the line it can get confusing as to why it's separated.

Sounds good! I'll be sure to group it more for future maps.

**Comment by csweet22** [comment]
> And can you change the name of "Entities" to "EnvironmentalInteractions". something to do with the environemnt.

I think Entities might make more sense. At least I wouldn't categorize Health Pickups and Enemy Spawners as being a part of the environment. The folder exists to separate those objects from the environment geometry.

**Pull request merged**